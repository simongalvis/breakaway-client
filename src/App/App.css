.App {
  text-align: center;
}
h1{
  float:left
}
