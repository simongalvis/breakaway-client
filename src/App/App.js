import React from 'react';
import {Route, Link } from 'react-router-dom';
//import config from '../config';
import './App.css';
import NavBar from '../NavBar/NavBar'
import LandingPage from '../LandingPage/LandingPage'
import LoginPage from '../LoginPage/LoginPage'
import SignupPage from '../SignupPage/SignupPage'
import Footer from '../Footer/Footer'
import ExpandedActivity from '../ExpandedActivity/ExpandedActivity'
import Dashboard from '../Dashboard/Dashboard'




class App extends React.Component {



  renderNavRoutes(){
    return(
      <>
    <Route exact path='/' component={NavBar}/>
    </>
    )
  }

  renderMainRoutes(){
    return(
      <>
    <Route exact path='/' component={LandingPage}/>
    <Route exact path='/signup-page' component={SignupPage}/>
    <Route exact path='/login-page' component={LoginPage}/>
    <Route exact path='/activities' component={Dashboard}/>
    <Route exact path='/activities/:activityId' component={ExpandedActivity}/>

</>
    
    )
  }

  render(){
  return (
    <div className="App">
      <nav>
        {this.renderNavRoutes()}
      </nav>

      <main>
        {this.renderMainRoutes()}
      </main>
      <Footer/>
      
    </div>
  );
  }
}

export default App;
