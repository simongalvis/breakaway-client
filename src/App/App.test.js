import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import renderer from 'react-test-renderer';

describe(`App component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<App/>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  }) 
  it(`renders UI as expected`, () =>{
    const tree = renderer
      .create(<App/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
