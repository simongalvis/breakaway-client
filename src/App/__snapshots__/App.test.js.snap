// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders UI as expected 1`] = `
<div
  className="App"
>
  <nav>
    <header>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Blue breakaway logo"
          id="main-logo"
          src="breakaway-logo.png"
        />
      </a>
      <a
        href="/login-page"
        onClick={[Function]}
      >
        <button
          className="nav-button"
          id="login-button"
          value="login"
        >
          Log In
        </button>
      </a>
    </header>
  </nav>
  <main>
    <div
      className="LandingPage"
    >
      <h2
        className="hero"
      >
        The simple way to track your athletic progress
      </h2>
      <p
        id="hero-text"
      >
        Breakaway allows you to log your runs, walks, hikes, and other activities in one convenient location
      </p>
      <img
        alt="Breakaway ui on mobile"
        id="landingpage-ui-image"
        src="sample-hero.jpg"
      />
      <a
        href="/signup-page"
        onClick={[Function]}
      >
        <button
          id="landingpage-signup-button"
        >
          <b>
            Sign Up
          </b>
        </button>
      </a>
    </div>
  </main>
  <footer>
    <p>
      Â© - 
      <a
        href="http://simongalvis.com/"
      >
        simongalvis.com
      </a>
    </p>
  </footer>
</div>
`;
